timestamp.go

```diff
@@ -22,28 +22,37 @@ type Timestamp time.Time
 // MarshalJSON defines how encoding/json marshals the object to JSON,
 // the result is a string of the UNIX timestamp
 func (t Timestamp) MarshalJSON() ([]byte, error) {
-	ts := t.Time().Unix()
+	var ts int64
+	if t.Time().Unix() < 1e12 {
+		// timestamp is in seconds
+		ts = t.Time().Unix()
+	} else {
+		// timestamp is in milliseconds
+		ts = t.Time().UnixMilli()
+	}
 	stamp := fmt.Sprint(ts)
-
 	return []byte(stamp), nil
+
 }
 
 // UnmarshalJSON defines how encoding/json unmarshals the object from JSON,
 // a UNIX timestamp string is converted to int which is used for the Timestamp
 // object value
 func (t *Timestamp) UnmarshalJSON(b []byte) error {
-	ts, err := strconv.Atoi(string(b))
+	if len(b) == 0 {
+		return nil
+	}
+	ts, err := strconv.ParseFloat(string(b), 64)
 	if err != nil {
 		return err
 	}
-
-	int64ts := int64(ts)
+	fmt.Println("b!!!", b)
```

Comment:
Please remove these prints before prodcution - we don't want Guy Carmi to hate us :)